### ISSUE TITLE
Examples integrating with umbrella elixir, phoenix, etc

### ISSUE THREAD

**Issue by aislanmaia** [maxohq/surrealix#3]
Would be great to have some examples, maybe some repos integrating this package with real elixir projects.

**Comment by jamilabreu** [comment]
+1 @mindreframer

**Comment by jamilabreu** [comment]
@aislanmaia found this demo repo https://github.com/maxohq/surrealix_demo, but more directed instructions would be amazing!

**Comment by mindreframer** [comment]
@jamilabreu Yeah, sorry about missing docs. In our project we actually decided to move away from SurrealDB for now and stick to plain Postgres. It solved all the performance troubles we had with SurrealDB with non-trivial amounts of data. Also the visibility and troubleshooting is simpler. Maybe SurrealDB becomes more usable for real-life projects in future, then we will invest more time into documenting the usage of Surrealix, for now this project is not in active use at our company

**Comment by lessless** [comment]
@mindreframer  what issue did you face with surreal? 
Would you advice to approach choosing SurrealDB as a primary data storage with caution?

**Comment by mindreframer** [comment]
@lessless we had large rather nested json payloads (JIRA issues), that we were storing in SurrealDB. Not too much, around 30k records. Query performance for single records with relations was good, but doing any searching / filtering / pagination was painfully slow (500ms or longer). Also the indexes at that time were not picked up by the query planner, so that it always had to do a full table scan for any "collections" query. 

Those issues were quite visible in Github issues, yet the team was focusing on implementing new features rather than fixing the basics. There were also a couple minor casting issues, documentation discrepancies and overall too much hype around things I personally do not need and less focus on storage basics. 
I would suggest generating a realistic amount of data for your app, dumping it into SurrealDB and running some plain queries  to see how it performs. Pay attention to the storage amount on the file system, memory usage during non-trivial queries, query execution times, handling of parallel-running queries. Do it with something simple, maybe use their Node.js SDK or similar. This is a throw-away project, do not over-engineer it. See how it works with low digit GBs, how long does it take to create indexes, are queries using them. A database that ignores indexes is mostly useless for real production cases. 

See https://surrealdb.com/docs/surrealdb/faqs/#is-surrealdb-ready-for-production-use. 

While mine intention was simplifying our stack, it ended up actually complicating it. Switching to Postgres resolved all query issues and I also realized that I did not need the flexibility of SurrealDB that much, as I imagined. Also the tooling around Postgres is so robust, that I could just focus on my app and not bother fixing DB issues. 

YMMV, for me it was clear, that it's too early for real production for SurrealDB. Maybe next year it will be more robust, until then this SDK is not maintained.

**Comment by lessless** [comment]
Thank you for the detailed feedback, @mindreframer. We have a reasonably large dataset, and our queries can take over a minute to execute without indexes. So, it's a no-go from the onset.

**Comment by lessless** [comment]
I'm periodically returning to look at SurrealDB because multimodal is important in making value out of data. 
I'm aware of only two multimodal relational/graph databases  - ArangoDB and SurrealDB.

Even though so far I'm landing with ArangoDB, there is every chance that SurrealDB may become a major booster for teams seeking an edge in providing more versatility to the business. Specifically historical queries over graphs is a killer feature. And I found out that the SurrealDB founder wrote a thesis exactly on that topic https://surrealdb.com/static/whitepaper.pdf. And  SurrealDB aims to provide it out of the box. Compare it with the application-level solution suggested by the ArangoDB team https://arangodb.com/learn/graphs/time-traveling-graph-databases/.

The problem is so complex that people are building frameworks to solve it: 
- https://github.com/RecallGraph/RecallGraph
- https://github.com/Bonsaya/arangodb-timetravel

Their storage that supports all of that is now in the beta stage https://surrealdb.com/features, and I think it might be worth giving it a shot after we are sure it doesn't have any serious bugs that can lead to data loss.

**Comment by mindreframer** [comment]
@lessless  I agree, that it looks very interesting on the paper. I took another looks one month ago, by downloading the recent version and started going through the __new__ tutorials/guides. 

Well, I hit very quickly couple discrepancies / bugs in the real behaviour vs documentation. And that is the thing that irks me. Either this is the "move fast and break things" mentality, which I do not appreciate in my database, or they are just somewhat sloppy in their documentation. 

Either way I do not have the mental capacity to reverse-engineer why documented behaviour is not matching the concrete implementation. Maybe Elixir spoiled me, but I expect basic documentation (nothing advanced really, I was just copy-pasting commands from their tutorials) to work as expected. 

Feel free to to fork the repo and try your luck, it seems your application would benefit from graph-like queries. I'm actually working with boring relational stuff, and yes, a working graph-engine would be awesome, but the emphasis is on "working". 

You also might consider looking into EdgeDB (now GelDB) - https://github.com/geldata/gel-elixir. This is a graph-like DB based on Postgres and they seem to be more stable than SurrealDB. The Elixir client is officially supported, has binary protocol, the team is really capable and focused on performance, the foundation is solid (Postgres), so... It might be a great alternative. 

Best, 
Roman
